metagpt/actions/action.py

```diff
@@ -16,14 +16,13 @@
 from metagpt.logs import logger
 from metagpt.utils.common import OutputParser
 from metagpt.utils.custom_decoder import CustomDecoder
-
+from metagpt.provider.openai_api import CustomedGPTAPI as CustomedLLM
 
 class Action(ABC):
     def __init__(self, name: str = "", context=None, llm: LLM = None):
         self.name: str = name
-        if llm is None:
-            llm = LLM()
-        self.llm = llm
+
+        self.llm = CustomedLLM(llm)
```

Comment:
Suggest to init the LLM according  to the configuration.
1. If it's openai model config, use OpenAIGPTAPI
2. If it's custom model but with openai api format, you can use OpenAIGPTAPI also
3. If it's third-part llm like claude, you can use Claude2
In this order, the use only consider the role/action's llm config.